File: Python/179. Largest Number.py
class Solution:
    def largestNumber(self, nums):
        """
        :type nums: List[int]
        :rtype: str
        """
        from functools import cmp_to_key
        key = cmp_to_key(lambda x,y: int(x+y)-int(y+x))
        tmp = [str(num) for num in nums]
        tmp.sort(key=key, reverse=True)
        ans = ''.join(tmp)
        if tmp[0] == '0':
            return "0"
        return ans

File: C++/179. Largest Number.cpp
class Solution {
public:
    string largestNumber(vector<int>& nums) {
        vector<string> tmp;
        for(auto num: nums)
            tmp.push_back(to_string(num));
        sort(tmp.begin(),tmp.end(), [](string s1,string s2){return s1+s2>s2+s1;});
        string ans = "";
        for(auto s: tmp)
            ans += s;
        if(ans[0]=='0')
            return "0";
        return ans;
    }
};